tests/haskell_module/multiple/BUILD.bazel

```
@@ -36,6 +37,18 @@ haskell_module(
     ],
 )
 
+haskell_library(
+    name = "lib",
+    # TODO: Remove when haskell_module supports dynamic builds
+    linkstatic = True,
+    modules = [
+        ":root",
+        ":branch_left",
+        ":branch_right",
+        ":leaf",
+    ],
+)
```

```suggestion
     deps = ["//tests/hackage:base"],
)
```
Same in the other cases.

I still think these should be considered direct dependencies of the `haskell_library|...`. Because they are linked against directly and because they are listed as direct dependencies in the resulting GHC package description.

With the comment before I meant that this should be enforced by `haskell_library|...`.

> I think it would be better if `haskell_library` complained because it is missing a dependency of its `haskell_module`s, namely `base`.


We can see if we want to change that once we have more clarity on the question posed in https://github.com/tweag/rules_haskell/issues/1552#issuecomment-961078202.

---

Done.